### DISCUSSION TITLE
Why does SWC treat a field called "mock" specially?

### DISCUSSION THREAD

**Discussion by fatso83** [swc-project/swc#8143] [Q&A] [answered]
I was writing [a how-to on using a new Sinon feature for auto-cleanup](https://github.com/sinonjs/sinon/pull/2556) of stubs injected using accessors when I discovered it did not work when I exposed accessors on an exported object called 'mock'. You can see the full diff here: https://github.com/fatso83/sinon-swc-bug/commit/eace1a762912e27975439b51baa64304c32321c1

Basically, if I write something like this:
```typescript
export const mock = {
    get toBeMocked(){ return toBeMocked; },
    set toBeMocked(mock){ toBeMocked = mock; }
}
``` 

it will be transpiled to this by SWC (very different from `tsc`!):
```typescript
export const _$mock = {
    get toBeMocked(){ return toBeMocked; },
    set toBeMocked(mock){ toBeMocked = mock; }
}
```

Somehow 'mock' gets turned into `_$mock`? Why does this happen? This breaks compatibility with Typescript as well, as TSC does not do this.

**Comment by kdy1** [comment]
It's because `jest`  mode is enabled and a variable named `mock` is special in jest

**Reply by fatso83** [reply]
I do not see any mentions of `jest` in the swcrc config? A bit strange if enabled by default.

```json
{
    "test": ".*.ts$",
    "jsc": {
      "parser": {
        "syntax": "typescript",
        "decorators": true
      },
      "transform": {
        "legacyDecorator": true,
        "decoratorMetadata": true
      }
    },
    "module": {
      "type": "commonjs"
    }
  }
```

I also have no dependencies on Jest
<details>

```json
  "devDependencies": {
    "@swc/cli": "^0.1.62",
    "@swc/core": "^1.3.71",
    "@swc/register": "^0.1.10",
    "@types/chai": "^4.3.5",
    "@types/mocha": "^10.0.1",
    "@types/node": "^20.4.4",
    "@types/sinon": "^10.0.15",
    "chai": "^4.3.7",
    "chai-as-promised": "^7.1.1",
    "mocha": "^10.2.0",
    "sinon": "^16.1.0"
  },
  "dependencies": {
    "ts-node": "^10.9.1",
    "typescript": "^5.1.6"
  }
```
</details>

Could you tell me how to disable that mode? I have searched for "swc jest mode", but I only end up on https://swc.rs/docs/usage/jest and similar pages mentioning this plugin, which is not used.

**Reply by kdy1** [reply]
It seems like you are using SWC via `@swc/jest`, right?
It's the reason and you can't disable it.
Why are you trying to do it?

**Reply by fatso83** [reply]
No. There is no Jest at all. No trace at all if grepping all the source code or lock files. See the deps: https://github.com/fatso83/sinon-swc-bug/blob/swc-special-handling-mock/package.json

**Reply by fatso83** [reply]
The tests run inside of Mocha

**Reply by kdy1** [reply]
Then please file an issue

**Reply by fatso83** [reply]
OK, I will! Was hoping it was not an issue and just me being stupid 😄

**Reply by fatso83** [reply, marked as answer]
Reported as #8148. It is actually ES5 as the target that will trigger the issue. Easy to see on [the playground](https://play.swc.rs/?version=1.3.94-nightly-20231016.1&code=H4sIAAAAAAAAA1WNQQrCMBBF93OKT1cteIPixp0LzyAlTmuwnZF0CkLJ3ZuIWrP9vP9ev4gzr4Kr6Ykv6h58qxusBAS2JQiqM7oJdmdo8IOXbkT%2FOVUtRSJ%2BPTUYRjbsDhz%2FjT%2FIqcyGKY0JyBFgKH51s37D%2B9giHt7sXLLZk%2FiimrfEU9wALRrz29oAAAA%3D&config=H4sIAAAAAAAAA1WPSw7DIAwF9zkF8rrbdtE79BAWdSIifrKJVBTl7iUE0maH3xsz8jooBbNoeKq1PMsQkYX4nEsi2Sf8lARIOxTNJia49XaWvRrRCtVoOxpIyBOluiX3hoMNQajjLXPGmzH%2FC3VwkUnkCu4o%2BsnSVTc0JbjwXmrZDkk50qF%2FwA%2FqsvNjMPLqm4kXGrYvhlQioBQBAAA%3D)
